pkg/tfbridge/x/token_test.go

```diff
@@ -169,6 +169,112 @@ func TestIgnored(t *testing.T) {
 	}, info.Resources)
 }
 
+func TestTokensInferredModules(t *testing.T) {
+	tests := []struct {
+		name            string
+		resourceMapping map[string]string
+		opts            *InferredModulesOpts
+	}{
+		{
+			name: "non-overlapping mapping",
+			resourceMapping: map[string]string{
+				"pkg_foo_bar":             "index:FooBar",
+				"pkg_fizz_buzz":           "index:FizzBuzz",
+				"pkg_resource":            "index:Resource",
+				"pkg_very_long_name":      "index:VeryLongName",
+				"pkg_very_very_long_name": "index:VeryVeryLongName",
+			},
+		},
+		{
+			name: "detect a simple module",
+			resourceMapping: map[string]string{
+				"pkg_hello_world":   "hello:World",
+				"pkg_hello_pulumi":  "hello:Pulumi",
+				"pkg_hello":         "hello:Hello",
+				"pkg_goodbye_folks": "index:GoodbyeFolks",
+				"pkg_hi":            "index:Hi",
+			},
+			opts: &InferredModulesOpts{
+				MinimumModuleSize: 2,
```

Comment:
A comment here highlighting what MinimumModuleSize is - I'm guessing this means that at least two entries are needed so "hello" got recognized as a module as it has 3 entries.